src/angular/app.js

```diff
@@ -147,6 +155,14 @@ controllers.controller('SearchCtrl', ['$scope', '$http', '$location', 'ServicesL
                 }
                 regions[region]++;
             }
+
+            var id = service.id;
+			if(id) {
+				if(servicesById[id] == null) {
+					servicesById[id] = service;
+				}
+//				servicesById[id]++;
+			}
```

Comment:
Do this initialization in the new `ServiceRepository`. Also check out your formatting since this looks weird. We're using 4 spaces for js files.